src/iregnet_fit.cpp

```diff
@@ -184,31 +207,103 @@ fit_cpp(Rcpp::NumericMatrix X, Rcpp::NumericMatrix y,
   }
   scale = scale_init; log_scale = log(scale);
 
+  /* Sort the whole matrix by censoring type, get the sorted X and y
+   * & Switch the right method for target_compute_grad_response
+   */
+  if(transformed_dist != IREG_DIST_EXTREME_VALUE){
+
+    bool sort = true;
+
+    if(transformed_dist == IREG_DIST_GAUSSIAN){
```

Comment:
Overall, I'm unsure about why we needed to split these functions for each distribution.

One the one hand, this causes the code to bloat up. Since the gradient equations are the same, I'm wondering if we could have reused more code.

On the other, maybe the optimizations were not possible without doing so? @tdhock do you have any thoughts about this?

Reply:
I'm not sure if it was necessary. IMO better to have slower, more maintainable code than bloated / duplicated / fast code. we should think about how to remove the duplication before merging.